eth/protocols/snap/sync.go

```diff
@@ -2409,10 +2490,19 @@ func (s *Syncer) forwardAccountTask(task *accountTask) {
 			return
 		}
 		task.Next = incHash(hash)
+
+		// Remove the completion flag once the account range is pushed
+		// forward. The leftover accounts will be skipped in the next
+		// cycle.
+		delete(task.stateCompleted, hash)
 	}
 	// All accounts marked as complete, track if the entire task is done
 	task.done = !res.cont
 
+	// Error out if there is any leftover completion flag.
+	if task.done && len(task.stateCompleted) != 0 {
+		log.Crit("Storage completion flags should be emptied", "number", len(task.stateCompleted))
```

Comment:
Here too I would either do panic or log.Error. I think log.Crit was a bad idea to have